ci-scripts/yocto2.groovy

```diff
@@ -247,37 +210,65 @@ void buildManifest(String variantName, String imageName, String layerToReplace="
         boolean bitbakeTests = getBoolEnvVar("BITBAKE_TEST", false)
         boolean yoctoCompatTest = getBoolEnvVar("YOCTO_COMPATIBILITY_TEST", false)
-        setupBitbake(yoctoDir, templateConf, doArchiveCache, smokeTests, analyzeImage)
-        setupCache(yoctoDir, yoctoCacheURL)
+        stage("Setup bitbake and cache") {
+            setupBitbake(yoctoDir, templateConf, doArchiveCache, smokeTests, analyzeImage)
+            setupCache(yoctoDir, yoctoCacheURL)
+        }
 
         // Build the images
         try {
+            stage("Bitbake ${imageName} for ${variantName}") {
+                buildImage(yoctoDir, imageName, variantName)
+            }
+
             boolean buildUpdate = variantName.startsWith("rpi") || variantName.startsWith("intel") || variantName.startsWith("arp")
-            buildImageAndSDK(yoctoDir, imageName, variantName, buildUpdate)
+            stage("Bitbake Update ${imageName} for ${variantName}") {
+                when (buildUpdate) {
+                    sh "/workspace/cookbook/yocto/build-images.sh ${yoctoDir} ${imageName}-update"
+                }
+            }
+
+            boolean buildSDK = getBoolEnvVar("BUILD_SDK", false)
+            stage("Build SDK ${imageName}") {
+                when (buildSDK) {
+                    sh "/workspace/cookbook/yocto/build-sdk.sh ${yoctoDir} ${imageName}"
+                }
+            }
+
             if (yoctoCompatTest) {
                 runYoctoCheckLayer(yoctoDir)
             }
             if (smokeTests) {
+                stage ("Run smoke tests")
                 runSmokeTests(yoctoDir, imageName)
-                if(bitbakeTests) {
-                    runBitbakeTests(yoctoDir)
+                stage("Run bitbake tests"){
+                    when (bitbakeTests) {
+                        runBitbakeTests(yoctoDir)
+                    }
                 }
             }
-
         } finally {
             // Archive cache even if there were errors.
-            archiveCache(yoctoDir, doArchiveCache, yoctoCacheArchivePath)
+            stage("Archive cache") {
+                when (doArchiveCache) {
+                    archiveCache(yoctoDir, doArchiveCache, yoctoCacheArchivePath)
+                }
+            }
 
             // Check if we want to store the images, SDK and artifacts as well
             boolean doArchiveArtifacts = getBoolEnvVar("ARCHIVE_ARTIFACTS", false)
-            if (doArchiveArtifacts) {
-                echo "ARCHIVE_ARTIFACTS was set"
-                archiveImagesAndSDK(yoctoDir, variantName)
+            stage("Archive build artifacts") {
+                when (doArchiveArtifacts) {
+                    echo "ARCHIVE_ARTIFACTS was set"
+                    archiveImagesAndSDK(yoctoDir, variantName)
+                }
             }
         }
 
     } finally {
-        deleteYoctoBuildDir("${yoctoDirInWorkspace}")
+        stage("Deleting Yocto build directory") {
```

Comment:
I'm picky again, sorry, but condsider to keep it short like `"Delete workspace"`

Reply:
It's not deleting workspace actually. It's deleting only build dir. Clean workspace here is a good idea, by the way

Reply:
Well, not sure.

There are actually occasions when I would like to access the workspace to read logs etc, especially if something goes wrong.

It would be good if the option to delete the workspace/build-dir could be an option, configurable from Jenkins job. Why have this built-in into the script at all?

Also note that build artifacts are put in build/tmp/deploy so there is a risk if the workspace is deleted too early. In my opinion, it should be done at the final step.

Reply:
It is a final step. So we delete build dir only after we have archived our artifacts